src/main/java/backend/Core.java

```diff
@@ -86,7 +87,8 @@ public Diff diff() throws IOException {
     Set<DiffEntry> deltaSource;
     Set<DiffEntry> deltaDoc;
 
-    String step = currentStep();
+    int tmp = Integer.parseInt(currentStep()) - 1;
+    String step = Integer.toString(tmp);
 
     try {
       deltaSource = repo.diffSinceTag(step);
```

Comment:
I don't know what happens here if we pass in a tag that doesn't exist. Thinking specifically when this is `"0"`

Reply:
Doesn't the if statement just before handle that? Or would we need to modify it to check if(t=null || t.equals("0"))?

Reply:
`currentStep()` will return `"1"` right after we init. Then we `- 1` at line 90, so we'll get `"0"`. Might be better to have a `lastStep()` method but that can wait for another PR, I think

Reply:
I'm still confused as to why this is a problem. Because after we init, if we run `g2t diff`, it would anyways throw a Null Pointer because there's nothing to compare with right. And for every step after it works normally. But we'll discuss when we meet next